src/index.ts

```diff
@@ -21,7 +21,13 @@ function decodeHtmlCharacters(str: string) {
 }
 
 function isValidUrl(url: string): boolean {
-  return URL.canParse(url);
+  try {
+    // We can replace this with URL.canParse() once it's more widely available
+    new URL(url); // eslint-disable-line no-new -- We're only checking for exceptions
```

Comment:
`return URL.canParse?.(url) ?? new URL(url) && true`

Reply:
This won't work, since `new URL(url)` throws an exception if it fails, so it needs to be wrapped in a `try..catch`.

But, we could do something like:

```ts
  if (URL.canParse) {
    return URL.canParse(url);
  }
  try {
    // We can replace this with URL.canParse() once it's more widely available
    new URL(url); // eslint-disable-line no-new -- We're only checking for exceptions
    return true;
  } catch (error) {
    return false;
  }
```

However, I'm not a fan of this. Sure, it's slightly faster in environments that do support `URL.canParse`, but the code is more complicated, and it's much harder to test, as we'd need environments that both support `URL.canParse` and don't support it.

Reply:
to be clear, I didn't suggest to remove the `try` catch

Reply:
I guess it's a useless change though, when the documentation says:

> It returns true for the same values for which the [URL() constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) would succeed, and false for the values that would cause the constructor to throw.

If that holds true, then don't worry about my suggestion